screens/CameraScreen.tsx

```diff
@@ -0,0 +1,277 @@
+import { useState, useCallback } from 'react';
+import { View, StyleSheet, TouchableOpacity, Text, Alert, Platform, StatusBar, SafeAreaView } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { CameraComponent } from '../components/Camera/CameraComponent';
+import { ImageGallery } from '../components/Camera/ImageGallery';
+import { MapViewComponent } from '../components/Map/MapViewComponent';
+import { MaterialIcons, Ionicons } from '@expo/vector-icons';
+import ImageView from 'react-native-image-viewing';
+import { Photo } from '../types';
+import { usePhotoStorage } from '../hooks/usePhotoStorage';
+import { PhotoViewer } from '../components/Camera/PhotoViewer';
+
+export default function CameraScreen() {
+  const insets = Platform.OS === 'android' ? useSafeAreaInsets() : { top: 0, bottom: 0 };
+  const {
+    photos,
+    addPhoto,
+    deletePhotoByIndex,
+    isLoading
+  } = usePhotoStorage();
+
+  const [showGallery, setShowGallery] = useState(false);
+  const [showMap, setShowMap] = useState(false);
+  const [selectedPhoto, setSelectedPhoto] = useState<Photo | null>(null);
+  const [isImageViewVisible, setIsImageViewVisible] = useState(false);
+  const [viewerVisible, setViewerVisible] = useState(false);
+  const [viewerIndex, setViewerIndex] = useState(0);
+
+  const handlePhotoTaken = useCallback(async (photo: Photo) => {
+    const savedPhoto = await addPhoto(photo);
+    if (!savedPhoto) {
+      Alert.alert('Error', 'Failed to save photo');
+    }
+  }, [addPhoto]);
+
+  const handleDeletePhoto = useCallback(async (index: number) => {
+    const success = await deletePhotoByIndex(index);
+    if (!success) {
+      Alert.alert('Error', 'Failed to delete photo');
+    }
+  }, [deletePhotoByIndex]);
+
+  const handlePhotoSelect = (photo: Photo) => {
+    const index = photos.findIndex(p => p.timestamp === photo.timestamp);
+    setViewerIndex(index >= 0 ? index : 0);
+    setViewerVisible(true);
+  };
+
+  const handleClosePhotoView = useCallback(() => {
+    setIsImageViewVisible(false);
+    setTimeout(() => setSelectedPhoto(null), 300);
+  }, []);
+
+  const toggleGallery = useCallback(() => {
+    setShowGallery(prev => !prev);
+    if (showMap) setShowMap(false);
+  }, [showMap]);
+
+  const toggleMap = useCallback(() => {
+    setShowMap(prev => !prev);
+    if (showGallery) setShowGallery(false);
```

Comment:
These two states seem to be coupled - any reason not to move them to the same `useState`?

Reply:
No, nothing very specific, it was just to try to complete it in less than 2 hours hehe, but with your suggestion we could have something like this

```suggestion
type ViewMode = 'camera' | 'gallery' | 'map';
    setShowMap(prev => !prev);
    if (showGallery) setShowGallery(false);

const [activeView, setActiveView] = useState<ViewMode>('camera');

const toggleView = useCallback((mode: ViewMode) => {
  setActiveView(prev => prev === mode ? 'camera' : mode);
}, []);
```